base_layer/wallet/src/error.rs

```diff
@@ -173,6 +173,75 @@ impl From<WalletStorageError> for ExitCodes {
+                matches!(self, WalletStorageError::BlockingTaskSpawnError(_))
+            },
+            WalletStorageError::FileError(_) => matches!(self, WalletStorageError::FileError(_)),
+            WalletStorageError::InvalidUnicodePath => matches!(self, WalletStorageError::InvalidUnicodePath),
+            WalletStorageError::HexError(_) => matches!(self, WalletStorageError::HexError(_)),
+            WalletStorageError::InvalidEncryptionCipher => matches!(self, WalletStorageError::InvalidEncryptionCipher),
+            WalletStorageError::InvalidPassphrase => matches!(self, WalletStorageError::InvalidPassphrase),
+            WalletStorageError::MissingNonce => matches!(self, WalletStorageError::MissingNonce),
+            WalletStorageError::AeadError(_) => matches!(self, WalletStorageError::AeadError(_)),
+            WalletStorageError::AlreadyEncrypted => matches!(self, WalletStorageError::AlreadyEncrypted),
+            WalletStorageError::ByteArrayError(_) => matches!(self, WalletStorageError::ByteArrayError(_)),
+            WalletStorageError::CannotAcquireFileLock => matches!(self, WalletStorageError::CannotAcquireFileLock),
+            WalletStorageError::DatabasePathIsRootPath => matches!(self, WalletStorageError::DatabasePathIsRootPath),
+            WalletStorageError::IoError(_) => matches!(self, WalletStorageError::IoError(_)),
+            WalletStorageError::NoPasswordError => matches!(self, WalletStorageError::NoPasswordError),
+            WalletStorageError::DeprecatedOperation => matches!(self, WalletStorageError::DeprecatedOperation),
+            WalletStorageError::KeyManagerError(_) => matches!(self, WalletStorageError::KeyManagerError(_)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn not_stack_overflow() {
+        #[allow(clippy::eq_op)]
+        let compare_true = WalletStorageError::InvalidUnicodePath == WalletStorageError::InvalidUnicodePath;
+        assert!(compare_true);
+
+        #[allow(clippy::eq_op)]
+        let compare_true = WalletStorageError::ValuesNotFound == WalletStorageError::ValuesNotFound;
+        assert!(compare_true);
+
+        let compare_false =
+            WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test1"))) ==
+                WalletStorageError::InvalidUnicodePath;
+        assert!(!compare_false);
+
+        let compare_true =
+            WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test1"))) ==
+                WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test2")));
+        assert!(compare_true);
+
+        let compare_true =
+            WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test1"))) ==
+                WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test1")));
```

Comment:
and
```suggestion
               let compare_true =  WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("test1")))==
                WalletStorageError::DieselR2d2Error(SqliteStorageError::DieselR2d2Error(String::from("something different")));
```
?

also true

Reply:
Is there a reason we need PartialEq implemented on the error?

Reply:
Removed these in total as per previous comment